### ISSUE TITLE
[Bug][a11y]Tapping new tab button should focus on address

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#16420]
## Prerequisites
Talkback should be enabled

## Steps to reproduce
1. Open the Tabs Tray
2. Tap the '+' new tab button

### Expected behavior
The focus should be set on the address bar

### Actual behavior
Focuses on the "Scan" button
or 
it focuses on the fill link from clipboard in case you have a link copied to clipboard.

### Device information

* Android device: 
• Google Pixel 3a (Android 11)
• Huawei Mate 20 Lite (Android 10)
• OnePlus A3 (Android 6.0.1)

* Fenix version: Nightly Build 201106
 
### Notes
Reproducible in both list view and grid view.
❗ Done previously in #11428

**Comment by sflorean** [comment]
Verified as fixed on Nightly 11/18 with Pixel 3(Android 11).
